fnlb/lb/allgrouper.go

```diff
@@ -117,69 +104,87 @@ func (a *allGrouper) remove(ded string) error {
 
 func (a *allGrouper) publishHealth() {
 
-	// get a list of healthy nodes
 	a.nodeLock.Lock()
-	new_list := make([]string, 0, len(a.nodeList))
+
+	// get a list of healthy nodes
+	newList := make([]string, 0, len(a.nodeList))
 	for key, value := range a.nodeList {
 		if value.healthy {
-			new_list = append(new_list, key)
+			newList = append(newList, key)
 		}
 	}
-	a.nodeLock.Unlock()
 
-	// sort and set the healty list pointer
-	sort.Strings(new_list)
-	a.healthyList.Store(&new_list)
+	// sort and update healthy List
+	sort.Strings(newList)
+	a.nodeHealthyList = newList
+
+	a.nodeLock.Unlock()
 }
 
 // return a copy
 func (a *allGrouper) List(string) ([]string, error) {
 
-	// safe atomic load on the current healthy list
-	ptr := a.healthyList.Load().(*[]string)
-	if len(*ptr) == 0 {
-		return nil, ErrNoNodes
-	}
+	a.nodeLock.RLock()
+	ret := make([]string, len(a.nodeHealthyList))
+	copy(ret, a.nodeHealthyList)
+	a.nodeLock.RUnlock()
 
-	ret := make([]string, len(*ptr))
-	copy(ret, *ptr)
-	return ret, nil
+	var err error
+	if len(ret) == 0 {
+		err = ErrNoNodes
```

Comment:
missed this the first go round -- I think returning an empty list here without an error is fine, the API uses this endpoint and it's not an error really, it's just an empty set.